### PR TITLE
LPS-83990 new portlet 3.0 cdi bean portlet project template

### PR DESCRIPTION
**Pull request by gamerson**

### CODE REVIEW

File: modules/sdk/project-templates/project-templates-war-cdi-bean-portlet/src/main/resources/archetype-resources/build.gradle [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+apply plugin: "com.liferay.css.builder"
+#if (${projectType} != "workspace")
+apply plugin: "war"
+#end
+
+dependencies {
+#if (${dependencyManagementEnabled} == "true")
+	compile group: "com.liferay", name: "com.liferay.bean.portlet.api"
+	compile group: "com.liferay", name: "com.liferay.bean.portlet.cdi.extension"
+	compile group: "org.jboss.weld.servlet", name: "weld-servlet"
```
Comment (petershin) [inline modules/sdk/project-templates/project-templates-war-cdi-bean-portlet/src/main/resources/archetype-resources/build.gradle]: Please add a new line between different dependency configurations (i.e. `compile` and `compileOnly`).

File: modules/sdk/project-templates/project-templates-war-cdi-bean-portlet/src/main/resources/archetype-resources/build.gradle [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	compileOnly group: "com.liferay.portal", name: "com.liferay.portal.kernel"
+	compileOnly group: "com.liferay.portal", name: "com.liferay.util.taglib"
+	compileOnly group: "javax.portlet", name: "portlet-api"
+	compileOnly group: "javax.servlet", name: "javax.servlet-api"
+	compileOnly group: "jstl", name: "jstl"
+#else
+	compile group: "com.liferay", name: "com.liferay.bean.portlet.api", version: "1.0.0"
+	compile group: "com.liferay", name: "com.liferay.bean.portlet.cdi.extension", version: "1.0.0"
+	compile group: "org.jboss.weld.servlet", name: "weld-servlet", version: "2.3.3.Final"
```
Comment (petershin) [inline modules/sdk/project-templates/project-templates-war-cdi-bean-portlet/src/main/resources/archetype-resources/build.gradle]: Please add a new line between different dependency configurations (i.e. `compile` and `compileOnly`).

File: modules/sdk/project-templates/project-templates-war-cdi-bean-portlet/src/main/resources/archetype-resources/src/main/java/portlet/__className__Portlet.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import javax.portlet.annotations.LocaleString;
+import javax.portlet.annotations.PortletConfiguration;
+import javax.portlet.annotations.RenderMethod;
+
+/**
+ * @author ${author}
+ */
+@PortletConfiguration(
+	portletName = ${className}PortletKeys.${className},
+	title = @LocaleString(value = ${className}PortletKeys.${className}))
```
Comment (petershin) [inline modules/sdk/project-templates/project-templates-war-cdi-bean-portlet/src/main/resources/archetype-resources/src/main/java/portlet/__className__Portlet.java]: Add a new line before the ending right parenthesis to match the existing pattern in portal.

File: modules/sdk/project-templates/project-templates-war-cdi-bean-portlet/src/main/resources/archetype-resources/src/main/java/portlet/__className__Portlet.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+@LiferayPortletConfiguration(
+	portletName = ${className}PortletKeys.${className},
+	properties = {
+		"com.liferay.portlet.display-category=category.sample",
+		"com.liferay.portlet.instanceable=true"
+	}
+)
```
Comment (petershin) [inline modules/sdk/project-templates/project-templates-war-cdi-bean-portlet/src/main/resources/archetype-resources/src/main/java/portlet/__className__Portlet.java]: If order doesn't matter, please move `@LiferayPortletConfiguration` before `@PortletConfiguration` so its sorted alphabetically.

File: modules/sdk/project-templates/project-templates-war-cdi-bean-portlet/src/main/resources/archetype-resources/src/main/java/portlet/__className__Portlet.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+public class ${className}Portlet {
+
+	@Inject
+	PortletConfig portletConfig;
```
Comment (petershin) [inline modules/sdk/project-templates/project-templates-war-cdi-bean-portlet/src/main/resources/archetype-resources/src/main/java/portlet/__className__Portlet.java]: If possible, add an explicit modifier for the `portletConfig` variable (i.e. `private PortletConfig _portletConfig;`).

File: modules/sdk/project-templates/project-templates-war-cdi-bean-portlet/src/main/resources/archetype-resources/src/main/java/portlet/__className__Portlet.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+	@RenderMethod(
+		include = "/WEB-INF/jsp/view.jsp",
+		portletNames = {${className}PortletKeys.${className}})
```
Comment (petershin) [inline modules/sdk/project-templates/project-templates-war-cdi-bean-portlet/src/main/resources/archetype-resources/src/main/java/portlet/__className__Portlet.java]: Add a new line before the ending right parenthesis to match the existing pattern in portal.

**Comment by petershin** [comment]
@gamerson - Closing this pull for now due to the publishing issue that we discussed on the call.

**Comment by ngriffin7a** [comment]
See https://github.com/petershin/liferay-portal/pull/182

**Pull request closed without merging**